arraycontext/fake_numpy.py

```diff
@@ -174,6 +174,53 @@ class BaseFakeNumpyLinalgNamespace:
     def __init__(self, array_context):
         self._array_context = array_context
 
+    def norm(self, ary, ord=None):
+        from numbers import Number
+        if isinstance(ary, Number):
+            return abs(ary)
+
+        if ord is None:
+            ord = 2
+
+        from arraycontext.impl import _is_meshmode_dofarray
+        from arraycontext.impl.pyopencl import _flatten_array
+
+        if _is_meshmode_dofarray(ary):
+            from warnings import warn
+            warn("Taking an actx.np.linalg.norm of a DOFArray is deprecated. "
+                    "(DOFArrays use 2D arrays internally, and "
+                    "actx.np.linalg.norm should compute matrix norms of those.) "
+                    "This will stop working in 2022. "
+                    "Use meshmode.dof_array.flat_norm instead.",
+                    DeprecationWarning, stacklevel=2)
+
+            import numpy.linalg as la
+            return la.norm(
+                    [self.norm(_flatten_array(subary), ord=ord)
```

Comment:
This is pyopencl specific

Reply:
Moved this part of the code to pyopencl.py